assets/src/pay-web/components/Breadcrumb.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from "react";
+import { Link } from "./Link";
+import styled from "../styled-components";
+import { useLocation } from "react-router-dom";
+
+const StyledBreadcrumbs = styled.nav`
+  font-size: 1.3rem;
+`;
+
+const flatMap = <X, Y>(fx: (value: X, index: number) => Y[], xs: X[]): Y[] =>
+  Array.prototype.concat(...xs.map(fx));
+
+// place a joiner between all other elements of an array
+const intersperse = <X,>(joiner: (index: number) => X, items: X[]): X[] => {
+  const interspersed = flatMap((item, index) => [item, joiner(index)], items);
+  interspersed.pop();
+  return interspersed;
+};
+
+// A link that renders as a span if it is the active route
+const Breadcrumb: React.FC<{ to?: string; children: string }> = ({
+  to,
+  children
+}) => {
+  const location = useLocation();
+
+  return !to || to === location.pathname ? (
+    <span>{children}</span>
+  ) : (
+    <Link to={to}>{children}</Link>
+  );
+};
+
+// Join together children with our caret icon
+const Breadcrumbs: React.FC<{}> = ({ children }) =>
+  React.Children.count(children) === 1 ? null : (
+    <StyledBreadcrumbs>
+      {intersperse(
+        n => (
+          <React.Fragment key={`caret-${n}`}>{" › "}</React.Fragment>
+        ),
+        React.Children.toArray(children)
+      )}
+    </StyledBreadcrumbs>
+  );
+
+// Add another breadcrumb to an existing set of breadcrumbs
+const createCrumbs = (
+  next: React.ReactNode,
+  crumbs: React.ReactNode[] = []
+): React.ReactNode[] => {
+  return [...crumbs, next];
+};
+
+interface SettingsParams {
```

Comment:
@dagoof the only issue with all the `*Params {}` interfaces here is that you are in the "@pay/web" module which is meant to be generic but you're pulling in domain stuff. I don't think it's a blocker to leave it in here, though it would be nice to keep it separate don't you think?

Reply:
yeah; good point. I'll move move it in subsequent PR.